server/src/server.ts

```diff
@@ -0,0 +1,125 @@
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import {
+    createConnection,
+    TextDocuments,
+    ProposedFeatures,
+    InitializeParams,
+    DidChangeConfigurationNotification,
+    SymbolKind,
+    SymbolInformation
+} from "vscode-languageserver";
+
+// Create a connection for the server. The connection uses Node"s IPC as a transport.
+// Also include all preview / proposed LSP features.
+const connection = createConnection(ProposedFeatures.all);
+
+// Create a simple text document manager. The text document manager
+// supports full document sync only
+const documents: TextDocuments = new TextDocuments();
+
+let hasConfigurationCapability: boolean = false;
+let hasWorkspaceFolderCapability: boolean = false;
+let hasDiagnosticRelatedInformationCapability: boolean = false;
+
+connection.onInitialize((params: InitializeParams) => {
+    let capabilities = params.capabilities;
+
+    // Does the client support the `workspace/configuration` request?
+    // If not, we will fall back using global settings
+    hasConfigurationCapability = !!(
+        capabilities.workspace && !!capabilities.workspace.configuration
+    );
+    hasWorkspaceFolderCapability = !!(
+        capabilities.workspace && !!capabilities.workspace.workspaceFolders
+    );
+    hasDiagnosticRelatedInformationCapability = !!(
+        capabilities.textDocument &&
+        capabilities.textDocument.publishDiagnostics &&
+        capabilities.textDocument.publishDiagnostics.relatedInformation
+    );
+
+    return {
+        capabilities: {
+            textDocumentSync: documents.syncKind,
+            documentSymbolProvider: true,
+        }
+    };
+});
+
+connection.onInitialized(() => {
+    if (hasConfigurationCapability) {
+        // Register for all configuration changes.
+        connection.client.register(DidChangeConfigurationNotification.type, undefined);
+    }
+    if (hasWorkspaceFolderCapability) {
+        connection.workspace.onDidChangeWorkspaceFolders(_event => {
+            connection.console.log("Workspace folder change event received.");
+        });
+    }
+});
+
+
+// TODO(Kelosky): in the future, perform diagnostic information (red squiggle) and listen
+// for configuration settings as in the LSP sample
+connection.onDocumentSymbol((parm) => {
+
+    const symbols: SymbolInformation[] = [];
+    const document = documents.get(parm.textDocument.uri);
+    if (!document) {
+        return null;
+    }
+
+    const lines = document.getText().split('\n');
+    for (let i = 0; i < lines.length - 1; i++) {
+
+        // if space or * in column one, it's not a symbol
+        if (lines[i][0] !== ' ' && lines[i][0] !== '*') {
+
+            // compress everything to one space
+            const tokenizedLine = lines[i].replace(/\s+/g, " ");
```

Comment:
You could have just done a string split here instead of parsing using dubstring down below.

```Typescript
const tokenizedLine = lines[i].replace(/\s+/g, " ").split();
```

You would then have an array of tokens that you address.

Reply:
thanks - ill fix this in the next PR.